app/src/main/java/com/example/sam/bucketlist/views/users/LoginActivity.java

```diff
@@ -0,0 +1,124 @@
+package com.example.sam.bucketlist.views.users;
+
+import android.app.Activity;
+import android.content.SharedPreferences;
+import android.graphics.Typeface;
+import android.os.Bundle;
+import android.content.Intent;
+import android.os.Handler;
+import android.preference.PreferenceManager;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.Button;
+import android.widget.ProgressBar;
+import android.widget.Toast;
+
+import com.example.sam.bucketlist.api.APIManager;
+import com.example.sam.bucketlist.models.UserFields;
+import com.example.sam.bucketlist.R;
+import com.example.sam.bucketlist.views.bucketlists.BucketlistActivity;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+
+public class LoginActivity extends Activity {
+
+    EditText userName,password;
+    Button login,forgot_pass;
+
+    ProgressBar progressBar;
+    private Handler handler = new Handler();
+
+    private int progressStatus = 0;
+
+
+    @Override
+    protected  void  onCreate(Bundle savedInstanceState){
+
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.login_activity_layout);
+
+
+        Typeface face=Typeface.createFromAsset(getAssets(),"fonts/Roboto-Regular.ttf");
+
+        userName = findViewById(R.id.uname);
+        userName.setTypeface(face);
+
+        password = findViewById(R.id.upass);
+        password.setTypeface(face);
+
+        login = findViewById(R.id.login);
+        login.setTypeface(face);
+
+
+        progressBar = findViewById(R.id.progressbar);
+        progressBar.setVisibility(View.INVISIBLE);
+
+        login.setOnClickListener(new View.OnClickListener() {
+
+            @Override
+            public void onClick(View view) {
+                APIManager bucketList = new APIManager(userName.getText().toString(),
+                        password.getText().toString());
+
+
+
+                 bucketList.login(new Callback<UserFields>() {
+
+                     @Override
+                     public void onResponse(Call<UserFields> call, Response<UserFields> response) {
+
+                         ProgressBar(progressBar);
+
+                         try {
+
+                             if (response.body().getToken().isEmpty()) {
+
+                                 Toast.makeText(LoginActivity.this, "Oops! Wring Credentials", Toast.LENGTH_SHORT).show();
+
+                             } else {
+                                 Toast.makeText(LoginActivity.this, "Super Dive " + response.body().getToken(), Toast.LENGTH_SHORT).show();
+
+                                 SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(LoginActivity.this);
+                                 sp.edit().putString("token", response.body().getToken()).apply();
+
```

Comment:
If possible create a shared preference class eg `BucketSharedPreferences` then add methods of setting and getting the token. Move the above logic to that class so that where you're retrieving or setting the token its easier with few lines of code.